### PR TITLE
LOOP-3064 PumpManager and DeviceManager protocol cleanup

### PR DESCRIPTION
**Pull request by ps2**
https://tidepool.atlassian.net/browse/LOOP-3064

These should all be just cleanup refactorings, and not affect behavior.

### CODE REVIEW

File: Loop/Managers/LoopDataManager.swift [hunk 1]
```diff
@@ -588,7 +588,7 @@ extension LoopDataManager {
     ///
     /// - Parameters:
     ///   - dose: The DoseEntry representing the confirmed bolus.
-    func bolusConfirmed(_ dose: DoseEntry, completion: (() -> Void)?) {
+    func bolusConfirmed(completion: (() -> Void)?) {
```
Comment (nhamming) [inline Loop/Managers/LoopDataManager.swift]: [nit] update parameter description

File: Loop/Managers/LoopDataManager.swift [hunk 2]
```diff
@@ -604,7 +604,7 @@ extension LoopDataManager {
     /// Notifies the manager that the bolus failed.
     ///
     /// - Parameters:
-    ///   - dose: The DoseEntry representing the confirmed bolus.
+    ///   - error: An error describing why the bolus request failed
```
Comment (nhamming) [inline Loop/Managers/LoopDataManager.swift:608]: [nit] add completion to the parameter description

File: Loop/Managers/LoopDataManager.swift [hunk 3]
```diff
@@ -1896,7 +1893,7 @@ protocol LoopDataManagerDelegate: class {
     ///   - basal: The new recommended basal
     ///   - completion: A closure called once on completion
     ///   - result: The enacted basal
-    func loopDataManager(_ manager: LoopDataManager, didRecommendBasalChange basal: (recommendation: TempBasalRecommendation, date: Date), completion: @escaping (_ result: Result<DoseEntry>) -> Void) -> Void
+    func loopDataManager(_ manager: LoopDataManager, didRecommendBasalChange basal: (recommendation: TempBasalRecommendation, date: Date), completion: @escaping (Error?) -> Void) -> Void
```
Comment (nhamming) [inline Loop/Managers/LoopDataManager.swift:1897]: [nit] update parameter description

**Review by nhamming — APPROVED** [review]
LGTM

**Review by rickpasetto — APPROVED** [review]
LGTM

**Pull request merged**